### PR TITLE
SRE White paper - openshift cluster

### PR DESCRIPTION
**Pull request by ShellyXueHan**

### CODE REVIEW

File: docs/white-paper.md [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+This section will discuss some of the high service level objectives that were identified by the PaaS between the DXE team and the Platform Team by environment. These are not set in stone as this is a living document that could change. 
+
+**Current Silver Cluster SLOs**
+•	90.0% availability for a single-node application deployment
+•	99.5% availability for multi-node application deployments
+
+**Current Gold Cluster SLOs**
+•	99.5% availability for multi-node application deployments
+
+**Current DXE Team SLO against the Cluster API Availability**
```
Comment (StevenBarre) [inline docs/white-paper.md]:
```suggestion
**Current DXC Team SLO against the Cluster API Availability**
```

File: docs/white-paper.md [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+Saturation – Identifying when services or the application is being access (aka queued). For example, if the service is an Application Program Interface (API) server such as Spring Boot and the Payload is taking a while to return because it is processing other requests. This can come in the form of utilization metrics. 
+Errors – These are Errors that are actually “reported” against the service or application. 
+Currently BC PaaS allows for the collection of the USE Method. Which can const of items such as:
+•	Resource utilization of Memory and CPU. 
+However, disk utilization I am currently unsure of as this is a on prem cluster and needs to be validated that the below query examples can be ran depending on permission issues from a client perspective. Below references some “Prometheus” queries that be used. Prometheus is one of the tool options that can be utilized. 
+These are the standard metrics identified for Product Team. Golden Signals is a good place to start. These are metrics that can be identified no matter what the application is. Let’s look into Golden Signals. 
+ 
+The above is a high-level graphic of metrics to use for USE Method
```
Comment (StevenBarre) [inline docs/white-paper.md:106]: What graphic?

File: docs/white-paper.md [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+```
+
+Memory Saturation:
+```
+1e3 * sum( rate(node_vmstat_pgpgin{job=“…”}[1m]) + rate(node_vmstat_pgpgout{job=“…”}[1m])) )
+```
+
+OpenShift Example:
+```
+max by (pod,namespace) (container_memory_working_set_bytes{container!=""} / on (pod,namespace,container)
```
Comment (StevenBarre) [inline docs/white-paper.md]: This query doesn't look right?

We don't have swap on our clusters, so there should be no paging of memory. Once its full it gets OOM Killed.

**Review by StevenBarre — CHANGES_REQUESTED** [review]
Looks like this was copied from a word/google doc? Needs a lot of formatting works still. I would recommend using VSCode with a few Markdown addons to help clean it up.

- https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint
- https://marketplace.visualstudio.com/items?itemName=yzhang.markdown-all-in-one

You can also use the View File option in GitHub to see how its rendered.
image

The contents of the doc however look good to me overall, just a few nit picks.

File: docs/white-paper.md [hunk 4]
```diff
@@ -0,0 +1,269 @@
+# SITE RELIABILITY ENGINEERING (SRE) AT BRITISH COLUMBIA WITHIN THE PLATFORM AS A SERVICE (PAAS) – WHITE PAPER
+
+This white paper is more of research analysis of British Columbia’s environment and identifying how to setup SRE related activities. This paper also provides some information about some common SRE principles and procedures. The paper will also provide a walk through of an example Product Team setting SRE objectives, indicators, monitoring, and observability with BC PaaS. 
+## IDENTIFICATION OF ENVIRONMENTS FOR THE PAAS
+It is important to notate that not all environments are created equal but if these cluster have core services or dependences then these need to be identified. Below is a list of environments currently identified for the BC PaaS.
+•	Gold Cluster
+•	Silver Cluster
+•	Emerald Cluster
+•	KLab Cluster
+•	KLab2 Cluster
```
Comment (mitovskaol) [inline docs/white-paper.md]:
```suggestion
•	KLab2 Cluster
•	CLab Cluster
```

File: docs/white-paper.md [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+
+**Product Team Environments:**
+•	Gold Cluster
+•	Silver Cluster
+•	Emerald Cluster
+•	ROSA Cluster – **In Progress**
+
+**Platform Team Environments:**
+•	KLab Cluster
+•	KLab2 Cluster
```
Comment (mitovskaol) [inline docs/white-paper.md]:
```suggestion
•	KLab2 Cluster
•	CLab Cluster
```

File: docs/white-paper.md [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+“We want to provide a Platform to our customers that allow highly availability, scalability, and reliability across British Columbia meeting the mission needs”. 
+IDENTIFY HIGH SERVICE LEVEL OBJECTIVES (SLO)S BY ENVIRONMENT
+This section will discuss some of the high service level objectives that were identified by the PaaS between the DXE team and the Platform Team by environment. These are not set in stone as this is a living document that could change. 
+
+**Current Silver Cluster SLOs**
+•	90.0% availability for a single-node application deployment
+•	99.5% availability for multi-node application deployments
+
+**Current Gold Cluster SLOs**
+•	99.5% availability for multi-node application deployments
```
Comment (mitovskaol) [inline docs/white-paper.md]:
```suggestion
•	99.95% availability for application deployments with a geographic failover
```

File: docs/white-paper.md (hunk 3 above)
Comment (StevenBarre) [reply docs/white-paper.md]: pagein / pageout are not necessarily related to swap, should be fine to leave in. https://serverfault.com/a/270324/27378

**Review by w8896699 — APPROVED** [review]
I have updated based on comment. Also I have create a ticket to do more research on cluster wide memory monitoring standard

**Pull request merged**